src/include/sof/lib/dai.h

```diff
@@ -252,23 +285,33 @@ static inline int dai_get_info(struct dai *dai, int info)
 		break;
 	}
 
+	platform_shared_commit(dai, sizeof(*dai));
+
 	return ret;
 }
 
 static inline void dai_write(struct dai *dai, uint32_t reg, uint32_t value)
 {
 	io_reg_write(dai_base(dai) + reg, value);
+
+	platform_shared_commit(dai, sizeof(*dai));
 }
 
 static inline uint32_t dai_read(struct dai *dai, uint32_t reg)
 {
-	return io_reg_read(dai_base(dai) + reg);
+	uint32_t val = io_reg_read(dai_base(dai) + reg);
+
+	platform_shared_commit(dai, sizeof(*dai));
+
+	return val;
 }
 
 static inline void dai_update_bits(struct dai *dai, uint32_t reg,
 				   uint32_t mask, uint32_t value)
 {
 	io_reg_update_bits(dai_base(dai) + reg, mask, value);
+
+	platform_shared_commit(dai, sizeof(*dai));
```

Comment:
I don't think these accessors modify the dai struct itself, is this commit needed?